### PR TITLE
Added Digamma, Erf, Erfc, Lgamma, TanhGrad

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

Added Digamma, Erf, Erfc, Lgamma, TanhGrad

## How was this patch tested?

added unit tests

### CODE REVIEW

**Comment by yangw1234** [comment]
pr validation pass

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/TensorNumeric.scala [hunk 1]
```diff
@@ -1075,6 +1108,16 @@ object TensorNumericMath {
         require(lower <= upper, "lower bound must be less or equal than upper bound")
         math.min(math.max(a, lower), upper)
       }
+
+      override def erf(a: Double): Double = org.apache.commons.math3.special.Erf.erf(a)
+
+      override def erfc(a: Double): Double = org.apache.commons.math3.special.Erf.erfc(a)
+
+      override def logGamma(a: Double): Double =
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/TensorNumeric.scala]: https://software.intel.com/node/32df281c-66cb-4846-b8bf-bc1f844828c7#32DF281C-66CB-4846-B8BF-BC1F844828C7

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/TensorNumeric.scala [hunk 2]
```diff
@@ -1075,6 +1108,16 @@ object TensorNumericMath {
         require(lower <= upper, "lower bound must be less or equal than upper bound")
         math.min(math.max(a, lower), upper)
       }
+
+      override def erf(a: Double): Double = org.apache.commons.math3.special.Erf.erf(a)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/TensorNumeric.scala]: https://software.intel.com/node/d62fb8af-02ff-463f-b4b0-909d08708290#D62FB8AF-02FF-463F-B4B0-909D08708290

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/TensorNumeric.scala [hunk 3]
```diff
@@ -1075,6 +1108,16 @@ object TensorNumericMath {
...
+
+      override def erfc(a: Double): Double = org.apache.commons.math3.special.Erf.erfc(a)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/TensorNumeric.scala]: https://software.intel.com/node/5e70e4a2-dc1a-4c59-b45b-64834eb4e6e6#5E70E4A2-DC1A-4C59-B45B-64834EB4E6E6

**Comment by yangw1234** [comment]
As discussed offline, just use the apache math libs for now.
pr validation pass

**Pull request merged**